src/components/Dashboard/CreateForm/CreateForm.jsx

```diff
@@ -0,0 +1,56 @@
+import React, { useState, useCallback, useMemo } from 'react';
+
+import ChefController from '../../Controllers/ChefController';
+import DishController from '../../Controllers/DishController';
+import RestaurantController from '../../Controllers/RestaurantController';
+
+import styles from './CreateForm.module.css';
+
+const controllerMap = {
+  chef: ChefController,
+  dish: DishController,
+  restaurant: RestaurantController,
+};
+
+const CreateForm = ({ onSave, onCancel, objectType }) => {
+  const initialNewItem = {};
+  const selectedController =  useMemo(()=>controllerMap[objectType],[objectType]) 
+  const [newItem, setNewItem] = useState({ ...initialNewItem });
+
+  const renderInputFields = useCallback(() => {
+
+    const handleChange = (e) => {
+      const { name, value } = e.target;
+      setNewItem((prevItem) => ({ ...prevItem, [name]: value }));
+    };
+
+    if (!selectedController) {
+      console.error('Unsupported object type:', objectType);
+      return null;
```

Comment:
remove he log, if we don't use an error we should use the handling error as a toast error